From 5b34e0ce08a727e1d7911a2e2f7b815ab4e3e290 Mon Sep 17 00:00:00 2001
From: Yongzhi Chen <ychena@apache.org>
Date: Fri, 30 Sep 2016 10:39:11 -0400
Subject: [PATCH 0773/1164] CDH-44088 HIVE-14784: Operation logs are disabled
 automatically if the parent directory does not
 exist. (Naveen Gangam via Yongzhi Chen)

Change-Id: I2ebf9af48604f59ff1c55f158f2b1a1cd9cceb6c
---
 .../hive/service/cli/operation/Operation.java      |   11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/service/src/java/org/apache/hive/service/cli/operation/Operation.java b/service/src/java/org/apache/hive/service/cli/operation/Operation.java
index fadeb8c..9644995 100644
--- a/service/src/java/org/apache/hive/service/cli/operation/Operation.java
+++ b/service/src/java/org/apache/hive/service/cli/operation/Operation.java
@@ -230,6 +230,17 @@ protected void createOperationLog() {
               operationLogFile.getAbsolutePath());
           operationLogFile.delete();
         }
+        if (!operationLogFile.getParentFile().exists()) {
+          LOG.warn("Operations log directory for this session does not exist, it could have been deleted " +
+              "externally. Recreating the directory for future queries in this session but the older operation " +
+              "logs for this session are no longer available");
+          if (!operationLogFile.getParentFile().mkdir()) {
+            LOG.warn("Log directory for this session could not be created, disabling " +
+                "operation logs: " + operationLogFile.getParentFile().getAbsolutePath());
+            isOperationLogEnabled = false;
+            return;
+          }
+        }
         if (!operationLogFile.createNewFile()) {
           // the log file already exists and cannot be deleted.
           // If it can be read/written, keep its contents and use it.
-- 
1.7.9.5

